src/firefly/js/visualize/saga/ChartsSync.js

```diff
@@ -34,9 +34,21 @@ export function* syncCharts() {
             case TablesCntlr.TABLE_SORT:
             case TablesCntlr.TABLE_NEW_LOADED:
                 const {tbl_id} = action.payload;
-                tableStatsState = flux.getState()[TableStatsCntlr.TBLSTATS_DATA_KEY];
-                if (has(tableStatsState, tbl_id)) {
-                    TableStatsCntlr.dispatchLoadTblStats(request);
+
+                // table statistics and histogram data do not change on table sort
+                if (action.type !== TablesCntlr.TABLE_SORT) {
+                    tableStatsState = flux.getState()[TableStatsCntlr.TBLSTATS_DATA_KEY];
+                    if (has(tableStatsState, tbl_id)) {
```

Comment:
I'd like to leave it as is: presence of tbl_id as an attribute indicates that we'd like to watch this table. tbl_id is omitted cleared when table is removed. If leftovers are present it indicates a problem.